src/dev/build/tasks/patch_native_modules_task.test.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ToolingLog,
+  ToolingLogCollectingWriter,
+  createAnyInstanceSerializer,
+  createAbsolutePathSerializer,
+} from '@osd/dev-utils';
+import { Build, Config } from '../lib';
+import { PatchNativeModules } from './patch_native_modules_task';
+
+const log = new ToolingLog();
+const testWriter = new ToolingLogCollectingWriter();
+log.setWriters([testWriter]);
+expect.addSnapshotSerializer(createAnyInstanceSerializer(Config));
+expect.addSnapshotSerializer(createAnyInstanceSerializer(ToolingLog));
+expect.addSnapshotSerializer(createAbsolutePathSerializer());
+
+jest.mock('../lib/download');
+jest.mock('../lib/fs', () => ({
+  ...jest.requireActual('../lib/fs'),
+  untar: jest.fn(),
+  gunzip: jest.fn(),
+}));
+
+const { untar } = jest.requireMock('../lib/fs');
+const { gunzip } = jest.requireMock('../lib/fs');
+const { download } = jest.requireMock('../lib/download');
+
+async function setup() {
+  const config = await Config.create({
+    isRelease: true,
+    targetAllPlatforms: false,
+    targetPlatforms: {
+      linux: false,
+      linuxArm: false,
+      darwin: false,
+    },
+  });
+
+  const build = new Build(config);
+
+  download.mockImplementation(() => {});
+
+  untar.mockImplementation(() => {});
+
+  gunzip.mockImplementation(() => {});
+
+  return { config, build };
+}
+
+beforeEach(() => {
+  testWriter.messages.length = 0;
+  jest.clearAllMocks();
+});
+
+it('patch_native_modules task downloads the correct package', async () => {
```

Comment:
nit: `patch_native_modules` is inferred because of the file name. You could also be specific about what `correct` means here.

Reply:
I change `patch_native_modules task` to `patch native modules task `. Seems its name already explain what it is doing. Then I specify correct to `correct platform` to show platform is the concern.